### PR TITLE
feat: add focus-trap-react to modal

### PR DESCRIPTION
**Pull request by RedTn**
@pradel 

Closes https://github.com/pradel/react-responsive-modal/issues/36

I'm having trouble writing enzyme tests that can handle tabbing/shift+tabbing focus, any guidance would be appreciated and I'd be happy to write tests after that.

### CODE REVIEW

**Comment by RedTn** [comment]
I can't seem to understand why dtslint is failing, I'm not experienced with typescript so the things I've been testing has been guess and check.

Would it be ok to tslint disable that warning that happens? I noticed it's even happening on the v3.3.0 branch

**Comment by pradel** [comment]
Thank you for the pr I will take a look at it soon!
Yeah I saw the dtslint tests are now failing don't know why 🤔 I will them :)

File: __tests__/modal.js [hunk 1]
```diff
@@ -376,3 +377,36 @@ describe('modal', () => {
     });
   });
 });
+
+describe('ConditionalWrap', () => {
```
Comment (pradel) [inline __tests__/modal.js]: If you can move the tests to a separate file would be great 👍

**Comment by RedTn** [comment]
> Btw I fixed the pipelines on master so if you merge the master branch into your branch and it should be good.

Sure! I'll merge master in.

> I tried the feature and it looks super cool!
> I noticed one bug, when I click on the overlay, I can't close the modal anymore. Can you please fix this behavior?

Yea I noticed this happening too but I think that behaviour is more closely aligned with the [WAI-ARIA](https://bitsofco.de/accessible-modal-dialog/) specs for modals. I can still look into fixing this but just wanted to bring it to your attention first

**Comment by pradel** [comment]
Amazing! Thanks for your work! Will release a new version soon 😄

**Pull request merged**